### PR TITLE
Add deleteuser API

### PR DESCRIPTION
**Pull request by saw-jan**
Adds delete user API to existing APIs. 
Takes `email` as param and deletes user by email (considered to be unique).
Usage:
`curl --location --request DELETE 'http://localhost:3000/auth/deleteuser' \
--header 'Content-Type: application/json' \
--data-raw '{
        "email": "test@test.com",
        "password": "test"
}'`

### CODE REVIEW

File: app/controllers/auth_controller.rb [hunk 1]
```diff
@@ -1,5 +1,5 @@
 class AuthController < ApplicationController
-  skip_before_action :authenticate_request, only: %i[login register verify_email generate_otp verify_otp] 
+  skip_before_action :authenticate_request, only: %i[login register verify_email generate_otp verify_otp delete_user]
```
Comment (koushikmohan1996) [inline app/controllers/auth_controller.rb:2]: Anyone can delete anyone's account now, Which should not be allowed. Only the authenticated user can delete his account

Comment (saw-jan) [reply app/controllers/auth_controller.rb:2]: Yes. I also was concerned about that.
So, it must be like:
- passing email & password
- and if authorized then delete.
Will it be a good implementation?

**Comment by saw-jan** [comment]
Still need too add `delete_user` function in `skip_before_action`.
But user can be deleted only if email and password is valid.

**Comment by koushikmohan1996** [comment]
Thanks :)

**Pull request merged**